### ISSUE TITLE
Update to ESLint v9

### ISSUE THREAD

**Issue by sosnovsky** [FlowCrypt/flowcrypt-browser#5713]
Migration guide - https://eslint.org/docs/latest/use/migrate-to-9.0.0

**Comment by sosnovsky** [comment]
some tips on migrating to v9 https://x.com/karlhorky/status/1792500811307622631

it also mentions that it's easier to migrate to ESLint flat config before starting migration to v9, I filed separate issue for config migration - https://github.com/FlowCrypt/flowcrypt-browser/issues/5728

**Comment by ioanmo226** [comment]
@sosnovsky When upgrading to ESLint v9, I encountered two major issues with packages:

1. **[eslint-plugin-import](https://github.com/import-js/eslint-plugin-import)**: This package doesn't yet support ESLint v9 and flat config. The solutions could be:
   - Wait for their official release.
   - Submit a PR (though this takes time).
   - Remove the `import/order` rule temporarily, which seems to be the simplest solution until their official release.
   - https://github.com/FlowCrypt/flowcrypt-browser/blob/2b2c2599c9258457aa2ac8da7fff8ce46cb8ed9f/eslint.config.mjs#L122

2. **[typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)**: This package doesn't support ESLint v9. However, there is an alpha version (latest is ^8.0.0-alpha.30) that supports ESLint v9. We might consider using this version even though it's not yet stable.
   - [Official ESLint v9 support notice](https://github.com/typescript-eslint/typescript-eslint/issues/8211)

**Comment by ioanmo226** [comment]
Let me know your thoughts

**Comment by sosnovsky** [comment]
Oh, I thought `typescript-eslint` added support for eslint v9 in their current release, then let's wait before `typescript-eslint` update

**Comment by ioanmo226** [comment]
You mean postpone our ESLint v9 update until `typescript-eslint` stable release and `eslint-plugin-import` release for ESLint v9 support?

**Comment by sosnovsky** [comment]
Yes, let's wait until support in stable typescript-eslint version. Maybe import plugin will add support too by that time, but it's not so critical.

**Comment by sosnovsky** [comment]
By the way, it looks like currently we don't use any rules from `eslint-plugin-import` - eslint config mentions only `'import/order'` rule, but it's `off`. I haven't found any other rules from this package in our config, so it should be ok to remove it.

**Comment by ioanmo226** [comment]
Sounds good!

**Comment by ioanmo226** [comment]
@sosnovsky While upgrading to ESLint v9, I encountered below eslint error and found this PR.
Do you wanna me to wait until official `eslint-plugin-header` release or use https://www.npmjs.com/package/@tony.ganchev/eslint-plugin-header patch version?

https://github.com/Stuk/eslint-plugin-header/pull/58

> Error: Key "rules": Key "header/header":
> 	Value ["block"," ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com "] should NOT have more than 0 items.
>

**Comment by sosnovsky** [comment]
What about using some alternative plugin - https://www.npmjs.com/package/eslint-plugin-license-header or https://www.npmjs.com/package/eslint-plugin-notice ? As https://www.npmjs.com/package/@tony.ganchev/eslint-plugin-header looks quite abandoned

**Comment by sosnovsky** [comment]
There are also compatibility utilities provided by ESLint - https://eslint.org/blog/2024/05/eslint-compatibility-utilities/#using-the-compatibility-utilities, maybe they can fix header plugin error?

**Comment by ioanmo226** [comment]
Let me check

**Comment by ioanmo226** [comment]
I just finished checking out [eslint-plugin-license-header](https://www.npmjs.com/package/eslint-plugin-license-header), and it's throwing a `Missing license header` error even though I've set the correct rules and header(Maybe it has some compatability issues).
Additionally, [@eslint/compat](https://npmjs.com/package/@eslint/compat) doesn't resolve header plugin issue and also throws the same error for `eslint-plugin-header`.

While [eslint-plugin-notice](https://github.com/nickdeis/eslint-plugin-notice) works well, it doesn't support auto-fixing. On the other hand, the current `eslint-plugin-header` does support auto-fixing with ESLint v8. For instance, when a user doesn't add a header and tries to save the file, it automatically fixes it by adding the correct header. Unfortunately, `eslint-plugin-notice` lacks this feature.

What's your take on this? Should we wait for the official release of `eslint-plugin-header`, use patch versions from forked repositories, or stick with `eslint-plugin-notice`, even though it doesn't support auto-fix?

**Comment by sosnovsky** [comment]
> What's your take on this? Should we wait for the official release of eslint-plugin-header, use patch versions from forked repositories, or stick with eslint-plugin-notice, even though it doesn't support auto-fix?

Thanks for checking it! Then let's use forked version for `eslint-plugin-header` for now, but it'll be better to create our own fork, so we won't depend on someone's else repository.

Hi @DenBond7, do you have permissions to create new repositories in `FlowCrypt` org? We need to create a fork for https://github.com/Stuk/eslint-plugin-header

**Comment by sosnovsky** [comment]
Thanks @DenBond7!

@ioanmo226 let's update this fork with `schema: false` fix and use it, until correct fix will be released in the original repo.

**Comment by ioanmo226** [comment]
Okay

**Comment by ioanmo226** [comment]
Just updated repository.
Do you wanna me to publish to npm with my npm account?

**Comment by sosnovsky** [comment]
Hmm, I think it's possible to use npm packages directly from github - https://stackoverflow.com/a/77764684/3091318

**Comment by ioanmo226** [comment]
Yeah, It's absolutely possible and it seems good to use repository directly until official release.
